### ISSUE TITLE
Add missing tests to CI

### ISSUE THREAD

**Issue by piersy** [clearmatics/autonity#464]
# Problem

The following tests are not being run by CI.

```
TestContractUpgradeSeveralUpgrades
TestContractUpgradeSeveralUpgradesOnBusTopology
TestContractUpgradeSeveralUpgradesOnStarTopology
TestContractUpgrade_Success
TestRemoveFromValidatorsList
TestTendermintAddConnectionToTopologySuccess
TestTendermintAddValidatorsToTopologySuccess
TestTendermintBusSuccess
TestTendermintChangeTopologyFromBusToStarSuccess
TestTendermintChangeTopologyFromStarToBusSuccess
TestTendermintNoQuorum
TestTendermintStarOverParticipantSuccess
TestTendermintStarSuccess
TestTendermintTC7
```

TestTendermintTC7 takes an incredibly long time to run.
`TestTendermintAddConnectionToTopologySuccess` is currently failing see #463 and is out of scope for this ticket.

The rest pass if run individually.

```
--- PASS: TestContractUpgradeSeveralUpgrades (24.87s)
--- PASS: TestContractUpgradeSeveralUpgradesOnBusTopology (25.00s)
--- PASS: TestContractUpgradeSeveralUpgradesOnStarTopology (25.00s)
--- PASS: TestContractUpgradeSeveralUpgradesOnBusTopology (24.53s)
--- PASS: TestContractUpgradeSeveralUpgradesOnStarTopology (25.58s)
--- PASS: TestContractUpgrade_Success (15.56s)
--- PASS: TestRemoveFromValidatorsList (15.40s)
--- PASS: TestTendermintAddValidatorsToTopologySuccess (56.54s)
    --- PASS: TestTendermintAddValidatorsToTopologySuccess/test_case_no_malicious (56.54s)
--- PASS: TestTendermintBusSuccess (9.55s)
    --- PASS: TestTendermintBusSuccess/test_case_no_malicious (9.55s)
--- PASS: TestTendermintChangeTopologyFromBusToStarSuccess (56.60s)
    --- PASS: TestTendermintChangeTopologyFromBusToStarSuccess/test_case_no_malicious (56.60s)
--- PASS: TestTendermintChangeTopologyFromStarToBusSuccess (56.66s)
    --- PASS: TestTendermintChangeTopologyFromStarToBusSuccess/test_case_no_malicious (56.66s)
--- PASS: TestTendermintNoQuorum (14.01s)
    --- PASS: TestTendermintNoQuorum/test_case_2_validators,_one_goes_down_after_block_3 (6.48s)
    --- PASS: TestTendermintNoQuorum/test_case_3_validators,_two_go_down_after_block_3 (7.53s)
--- PASS: TestTendermintStarOverParticipantSuccess (9.99s)
    --- PASS: TestTendermintStarOverParticipantSuccess/test_case_no_malicious (9.99s)
--- PASS: TestTendermintStarSuccess (9.53s)
    --- PASS: TestTendermintStarSuccess/test_case_no_malicious (9.53s)
```

 Although if run together some fail and they all fail due to the following error.
`leaks.go:78: found unexpected goroutines:`

```
--- PASS: TestRemoveFromValidatorsList (14.82s)
--- PASS: TestContractUpgrade_Success (15.00s)
--- PASS: TestContractUpgradeSeveralUpgrades (25.00s)
--- PASS: TestContractUpgradeSeveralUpgradesOnBusTopology (24.99s)
--- PASS: TestContractUpgradeSeveralUpgradesOnStarTopology (26.03s)
--- PASS: TestTendermintNoQuorum (12.97s)
    --- PASS: TestTendermintNoQuorum/test_case_2_validators,_one_goes_down_after_block_3 (4.96s)
    --- PASS: TestTendermintNoQuorum/test_case_3_validators,_two_go_down_after_block_3 (8.01s)
--- FAIL: TestTendermintStarSuccess (10.48s)
    --- FAIL: TestTendermintStarSuccess/test_case_no_malicious (10.48s)
--- FAIL: TestTendermintStarOverParticipantSuccess (11.01s)
    --- FAIL: TestTendermintStarOverParticipantSuccess/test_case_no_malicious (11.01s)
--- FAIL: TestTendermintBusSuccess (10.97s)
    --- FAIL: TestTendermintBusSuccess/test_case_no_malicious (10.96s)
--- FAIL: TestTendermintChangeTopologyFromBusToStarSuccess (59.01s)
    --- FAIL: TestTendermintChangeTopologyFromBusToStarSuccess/test_case_no_malicious (59.01s)
--- FAIL: TestTendermintChangeTopologyFromStarToBusSuccess (58.97s)
    --- FAIL: TestTendermintChangeTopologyFromStarToBusSuccess/test_case_no_malicious (58.97s)
--- FAIL: TestTendermintAddValidatorsToTopologySuccess (53.01s)
    --- FAIL: TestTendermintAddValidatorsToTopologySuccess/test_case_no_malicious (53.01s)
``` 

# Acceptance Criteria
The following tests should not fail if run together
```
TestTendermintStarSuccess
TestTendermintStarOverParticipantSuccess
TestTendermintBusSuccess
TestTendermintChangeTopologyFromBusToStarSuccess
TestTendermintChangeTopologyFromStarToBusSuccess
TestTendermintAddValidatorsToTopologySuccess
```

The following tests should be added to CI
```
TestContractUpgradeSeveralUpgrades
TestContractUpgradeSeveralUpgradesOnBusTopology
TestContractUpgradeSeveralUpgradesOnStarTopology
TestContractUpgrade_Success
TestRemoveFromValidatorsList
TestTendermintAddValidatorsToTopologySuccess
TestTendermintBusSuccess
TestTendermintChangeTopologyFromBusToStarSuccess
TestTendermintChangeTopologyFromStarToBusSuccess
TestTendermintNoQuorum
TestTendermintStarOverParticipantSuccess
TestTendermintStarSuccess
```

**Comment by yazzaoui** [comment]
I think goleak is counting the difference on the number of goroutines globally between the moment the test is starting until it finishes. Might explain why it fails when running together.
On top of that with our previous CI system, tests were flaky as it was very resource-intensive and slow to run multiple scenarios in parallel on the same VM, hence the reason the run them sequentially.
